modules/microvm/host/microvm-host.nix

```diff
@@ -122,22 +126,31 @@ in
         ++ vmRootDirs
         ++ xdgRules;
 
-      # TODO: remove hardcoded paths
       systemd.services =
         {
           # Generate anonymous unique device identifier
           generate-device-id = {
             enable = true;
-            description = "Generate device unique id";
+            description = "Generate device and machine ids";
             wantedBy = [ "local-fs.target" ];
             after = [ "local-fs.target" ];
             unitConfig.ConditionPathExists = "!/persist/common/device-id";
             serviceConfig = {
               Type = "oneshot";
-              ExecStart = "${pkgs.writeShellScript "generate-device-id" ''
-                # Generate a unique device id for the device
-                echo -n "$(od -txC -An -N6 /dev/urandom | tr ' ' - | cut -c 2-)" > /persist/common/device-id
-              ''}";
+              ExecStart =
+                [
```

Comment:
Should these guard, if the file exists? Not sure the use case to change every boot